File: spring_api/src/main/java/com/stressdebugger/model/StressLog.java
package com.stressdebugger.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "stress_logs")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StressLog {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String username;
    
    @Column(nullable = false, length = 1000)
    private String text;
    
    @Builder.Default
    private String logType = "NORMAL";
    
    private int angerLevel;
    private int anxiety;
    private int techFactor;
    private int humanFactor;
    
    @Column(length = 2000)
    private String forensicResult;
    
    @Column(length = 2000)
    private String justification;
    
    @Column(length = 2000)
    private String consolation;
    
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: flutter_app/lib/models/stress_log.dart
class StressLog {
  final int? id;
  final String text;
  final int? angerLevel;
  final int? anxiety;
  final int? techFactor;
  final int? humanFactor;
  final Map<String, dynamic>? forensicResult;
  final String? justification;
  final String? consolation;
  final DateTime? createdAt;

  StressLog({
    this.id,
    required this.text,
    this.angerLevel,
    this.anxiety,
    this.techFactor,
    this.humanFactor,
    this.forensicResult,
    this.justification,
    this.consolation,
    this.createdAt,
  });

  factory StressLog.fromJson(Map<String, dynamic> json) {
    return StressLog(
      id: json['id'],
      text: json['text'],
      angerLevel: json['angerLevel'],
      anxiety: json['anxiety'],
      techFactor: json['techFactor'],
      humanFactor: json['humanFactor'],
      forensicResult: json['forensicResult'],
      justification: json['justification'],
      consolation: json['consolation'],
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'text': text,
    };
  }
}
